src/sinks/datadog/logs/sink.rs

```diff
@@ -102,6 +103,47 @@ fn normalize_event(event: &mut Event) {
         log.rename_key(host_path, event_path!("hostname"));
     }
 
+    if let Some(ddtags_path) = log.find_key_by_meaning("tags") {
+        let ddtags_path = ddtags_path.clone();
+
+        // first, if the value is an array we need to reconstruct it to a comma delimited string for DD logs intake.
+        if let Some(Value::Array(tags_arr)) = log.get(&ddtags_path) {
+            if !tags_arr.is_empty() {
+                let mut all_tags = String::new();
+                tags_arr.iter().enumerate().for_each(|(idx, tag_kv)| {
+                    if let Some(bytes) = tag_kv.as_bytes() {
+                        if idx > 0 {
+                            all_tags.push(',');
+                        }
+                        let tag_kv_str = String::from_utf8_lossy(bytes);
+                        all_tags.push_str(&tag_kv_str);
+                    }
+                });
+                log.insert(&ddtags_path, all_tags);
+            }
```

Comment:
Optional suggestion:

```suggestion
        if !tags_arr.is_empty() {
            let all_tags: String = tags_arr.iter().filter_map(|tag_kv| {
                if let Some(bytes) = tag_kv.as_bytes() {
                    Some(String::from_utf8_lossy(bytes).to_string())
                } else {
                    None // Skip invalid UTF-8 bytes
                }
            }).collect::<Vec<String>>().join(",");
            log.insert(&ddtags_path, all_tags);
        }
```

I didn't test the above but it looks slightly cleaner.

Reply:
Nice, yeah I like that better. More clear.